File: php/src/Domain/Purchase/PurchasedBook.php
<?php

namespace Katalogs\BookInvoicing\Domain\Purchase;

use Katalogs\BookInvoicing\Domain\Books\Book;

class PurchasedBook
{
    public function __construct(
        public readonly Book $book,
        public readonly int $quantity
    ) { }

    public function totalPrice(): float
    {
        return $this->book->price * $this->quantity;
    }
}

File: go/purchase/purchased_book.go
package purchase

import . "book_invoicing/domain/book"

type PurchasedBook struct {
	Book     Book
	Quantity int
}

func NewPurchasedBook(book Book, quantity int) PurchasedBook {
	return PurchasedBook{
		Book:     book,
		Quantity: quantity,
	}
}

func (pb PurchasedBook) TotalPrice() float64 {
	return pb.Book.GetPrice() * float64(pb.Quantity)
}
